Framework/HistogramData/test/HistogramTest.h

```diff
@@ -1108,6 +1108,15 @@ class HistogramTest : public CxxTest::TestSuite {
     TS_ASSERT(!h.sharedE());
     TS_ASSERT(!h.sharedDx());
   }
+
+  void test_that_can_iterate_hsitogram() {
+    Histogram hist(Points{0.1, 0.2, 0.4}, Counts{1, 2, 4});
+    double total = 0;
+    for (const auto &item : hist) {
+      total += item.counts();
+    }
+    TS_ASSERT_EQUALS(total, 7)
+  }
```

Comment:
Test also:
- `begin()` and `end()` correct for both cases, point-histograms and normal histograms.
- Iterator comparison.
- Adding/subtracting from iterators.
- More?